src/code/z_lifemeter.c

```diff
@@ -1,59 +1,139 @@
+
+s16 sHeartPrimCurrent1[3];
+s16 sHeartEnvCurrent1[3];
+s16 sHeartPrimCurrent2[2][3];
+s16 sHeartEnvCurrent2[2][3];
+
+void Health_Init(GlobalContext* globalCtx) {
     InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
 
     interfaceCtx->unk_228 = 0x140;
     interfaceCtx->unk_226 = gSaveContext.health;
     interfaceCtx->unk_22A = interfaceCtx->unk_1FE = 0;
     interfaceCtx->unk_22C = interfaceCtx->unk_200 = 0;
 
-    interfaceCtx->unk_20E[0] = 255;
-    interfaceCtx->unk_20E[2] = 70;
-    interfaceCtx->unk_20E[4] = 50;
-    interfaceCtx->unk_21A[0] = 50;
-    interfaceCtx->unk_21A[2] = 40;
-    interfaceCtx->unk_21A[4] = 60;
-
-    interfaceCtx->unk_20E[1] = 255;
-    interfaceCtx->unk_20E[3] = 70;
-    interfaceCtx->unk_20E[5] = 50;
-    interfaceCtx->unk_21A[1] = 50;
-    interfaceCtx->unk_21A[3] = 40;
-    interfaceCtx->unk_21A[5] = 60;
-
-    D_8015FDD0[0] = D_8015FDD0[3] = 255;
-    D_8015FDD0[1] = D_8015FDD0[4] = 255;
-    D_8015FDD0[2] = D_8015FDD0[5] = 255;
-    D_8015FDE0[0] = D_8015FDE0[3] = 200;
-    D_8015FDE0[1] = D_8015FDE0[4] = 0;
-    D_8015FDE0[2] = D_8015FDE0[5] = 0;
+    interfaceCtx->healthPrimR[0] = HEART_PRIM_R;
+    interfaceCtx->healthPrimG[0] = HEART_PRIM_G;
+    interfaceCtx->healthPrimB[0] = HEART_PRIM_B;
+
+    interfaceCtx->healthEnvR[0] = HEART_ENV_R;
+    interfaceCtx->healthEnvG[0] = HEART_ENV_G;
+    interfaceCtx->healthEnvB[0] = HEART_ENV_B;
+
+    interfaceCtx->healthPrimR[1] = HEART_PRIM_R;
+    interfaceCtx->healthPrimG[1] = HEART_PRIM_G;
+    interfaceCtx->healthPrimB[1] = HEART_PRIM_B;
+
+    interfaceCtx->healthEnvR[1] = HEART_ENV_R;
+    interfaceCtx->healthEnvG[1] = HEART_ENV_G;
+    interfaceCtx->healthEnvB[1] = HEART_ENV_B;
+
+    sHeartPrimCurrent2[0][0] = sHeartPrimCurrent2[1][0] = HEART_DOUBLE_DEFENSE_PRIM_R;
+    sHeartPrimCurrent2[0][1] = sHeartPrimCurrent2[1][1] = HEART_DOUBLE_DEFENSE_PRIM_G;
+    sHeartPrimCurrent2[0][2] = sHeartPrimCurrent2[1][2] = HEART_DOUBLE_DEFENSE_PRIM_B;
+    sHeartEnvCurrent2[0][0] = sHeartEnvCurrent2[1][0] = HEART_DOUBLE_DEFENSE_ENV_R;
+    sHeartEnvCurrent2[0][1] = sHeartEnvCurrent2[1][1] = HEART_DOUBLE_DEFENSE_ENV_G;
+    sHeartEnvCurrent2[0][2] = sHeartEnvCurrent2[1][2] = HEART_DOUBLE_DEFENSE_ENV_B;
 }
 
 #ifdef NON_MATCHING
-// this function still needs some work but it should be functionally equivalent
-void Health_UpdateData(GlobalContext* globalCtx) {
+// Far from matching, but is equivalent. The for loop needs to become unrolled somehow in order to match.
+void Health_Update(GlobalContext* globalCtx) {
+    static s16 sHeartPrimColors[3][3] = {
+        { HEART_PRIM_R, HEART_PRIM_G, HEART_PRIM_B },
+        { HEART_BURN_PRIM_R, HEART_BURN_PRIM_G, HEART_BURN_PRIM_B },    // unused
+        { HEART_DROWN_PRIM_R, HEART_DROWN_PRIM_G, HEART_DROWN_PRIM_B }, // unused
+    };
+    static s16 sHeartEnvColors[3][3] = {
+        { HEART_ENV_R, HEART_ENV_G, HEART_ENV_B },
+        { HEART_PRIM_BURN_ENV_R, HEART_PRIM_BURN_ENV_G },            // unused
```

Comment:
did making this in function improve matching? otherwise it should prob be outside